### PR TITLE
Enhancement UploadField uses injector for child components

### PR DESCRIPTION
**Pull request by flamerohr**
Requires https://github.com/silverstripe/silverstripe-admin/pull/394

Fixes https://github.com/silverstripe/silverstripe-asset-admin/issues/718 to become more extensible

### CODE REVIEW

File: client/src/components/UploadField/tests/UploadField-test.js [hunk 1]
```diff
@@ -1,7 +1,5 @@
 /* global jest, jasmine, describe, it, expect, beforeEach */
 
-jest.mock('containers/InsertMediaModal/InsertMediaModal');
```
Comment (flamerohr) [inline client/src/components/UploadField/tests/UploadField-test.js]: Having removed this line - I really like to option to pass a prop to override the injected component :D

**Comment by unclecheese** [comment]
Have rebased to the `GalleryToolbar` changes.

**Comment by flamerohr** [comment]
Experiencing some (heavy) loading issues with this now...

We've tracked down the issue to composing every render which is cache busting and re-instantiating almost every component per cycle 💥 

Fix found here https://github.com/silverstripe/silverstripe-admin/pull/405

**Comment by unclecheese** [comment]
Behat test is failing on missing file history form. I can't replicate it locally, so I think it just needs a `    I wait for 2 seconds until I see the "#Form_fileHistoryForm" element`

**Comment by flamerohr** [comment]
Found this bug in cms causing the behat to fail https://github.com/silverstripe/silverstripe-cms/pull/2079

**Comment by unclecheese** [comment]
Merged. Rebasing.

**Comment by flamerohr** [comment]
All green :D

**Pull request merged**